internal/rpc/jsonrpc/methods.go

```diff
@@ -1216,13 +1224,359 @@ func (s *Server) getBlockHash(ctx context.Context, icmd interface{}) (interface{
 	return info.Hash.String(), nil
 }
 
+// getBlock implements the getblock command.
+func (s *Server) getBlock(ctx context.Context, icmd interface{}) (interface{}, error) {
+	cmd := icmd.(*types.GetBlockCmd)
+	w, ok := s.walletLoader.LoadedWallet()
+	if !ok {
+		return nil, errUnloadedWallet
+	}
+	n, err := w.NetworkBackend()
+	if err != nil {
+		return nil, err
+	}
+
+	blockHash, err := chainhash.NewHashFromStr(cmd.Hash)
+	if err != nil {
+		return nil, rpcError(dcrjson.ErrRPCDecodeHexString, err)
+	}
+
+	blocks, err := n.Blocks(ctx, []*chainhash.Hash{blockHash})
+	if err != nil {
+		return nil, err
+	}
+	if len(blocks) == 0 {
+		// Should never happen but protects against a possible panic on
+		// the following code.
+		return nil, rpcErrorf(dcrjson.ErrRPCInternal.Code, "Network returned 0 blocks")
+	}
+
+	blk := blocks[0]
+
+	// When the verbose flag isn't set, simply return the
+	// network-serialized block as a hex-encoded string.
+	if cmd.Verbose != nil && !*cmd.Verbose {
+		blkBytes, err := blk.Bytes()
+		if err != nil {
+			return nil, rpcErrorf(dcrjson.ErrRPCInternal.Code, "Could not serialize block: %v", err)
+		}
+		return hex.EncodeToString(blkBytes), nil
```

Comment:
let's avoid calling Bytes.  this should instead use a hex.Encoder.  there are some other examples of this technique in this file but this is the gist:

```
        b := new(strings.Builder)
        b.Grow(2 * blk.SerializeSize())
        err = blk.Serialize(hex.NewEncoder(b))
        return b.String(), nil
```